### PR TITLE
Begin new examples in getting started guides

### PR DESCRIPTION
**Pull request by ChrisChinchilla**

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
Hello all lovely ecosystem developers!

Well, specifically @realityking @Khaledgarbaya @mariobodemann and @dlitvakb 

I have been working on replacing examples in the documentation to use examples from the product catalog space. So far I've only reviewed Android, PHP, Ruby, and JavaScript, and so far, just the getting started guides for CDA, You can probably see the files I've worked on in the log.

This is not going to be merged yet, so don't worry too much about the space ids, content type ids etc, they may not stay like that, focus on the explanations and if the code is a good explanation.

You will notice that I am currently using hercule to use partials in the docs, allowing for the re-using of common elements. You can either read the docs on the normal folders and not worry too much about this more conceptual content or read the fully rendered content [here](https://github.com/contentful/slash-developers/tree/review/new-examples/out) (that may not stay in the repo in the long run).

Equally, I have added partials for the expected output of code, I'm not sure what we'll do with this right now, maybe nothing, maybe a toggle to see it, but you can review those or not if you want.

So in summary, whilst this PR looks huge, I'm asking for your input on one file each really, relevant to your languages.

**Comment by realityking** [comment]
Haven't looked completely through it but a quick observation: the leading `<?php` on PHP code snippets is there because the marketing-website needs it for proper syntax highlighting. In theory the syntax highlighter has an option not to need it but I never got that working.

File: docs/android/tutorials/getting-started-with-contentful-and-android.md [hunk 1]
```diff
@@ -13,94 +13,121 @@ nextsteps:
...
+:[Getting started tutorial intro](../../_partials/getting-started-intro.md)
 
 ## Create a new Android project
 
 Create a new project in Android Studio using the 'Blank Activity' template, and name it whatever you wish.
 
-## Define Contentful as a dependency
+## Dependencies and permissions
+
+This guide uses [RXAndroid](https://github.com/ReactiveX/RxAndroid) in the examples, which adds more complexity, but allows you to fetch results without tying up the main Android thread.
```
Comment (mariobodemann) [inline docs/android/tutorials/getting-started-with-contentful-and-android.md]: Shall we say it makes it more complicated, or should we just say it allows you to do stuff outside the main thread?

File: docs/android/tutorials/getting-started-with-contentful-and-android.md [hunk 2]
```diff
@@ -13,94 +13,121 @@ nextsteps:
...
 
+  <uses-permission android:name="android.permission.INTERNET" />
+  ...
 </manifest>
 ~~~
 
-Add the following code to the click handler of the floating action button, or in the `onCreate` function:
+## Creating a client
+
+Add the following code to the `onCreate` function to create a `CDAClient` that communicates with the Contentful APIs:
```
Comment (mariobodemann) [inline docs/android/tutorials/getting-started-with-contentful-and-android.md]: `function` → `method`

File: docs/android/tutorials/getting-started-with-contentful-and-android.md [hunk 3]
```diff
@@ -13,94 +13,121 @@ nextsteps:
...
-        .build();
+  .setSpace("<space_id>")
+  .setToken("<access_token>")
+  .build();
 ~~~
 
-This will create a `CDAClient` that communicates with the Contentful APIs using the space id and token you gained earlier.
-
-To fetch all entries:
+Add the [Gson](https://github.com/google/gson) library to make JSON responses easier to read.
```
Comment (mariobodemann) [inline docs/android/tutorials/getting-started-with-contentful-and-android.md]: Hmm, technically it is not about `adding` a library, since it comes already bundled in the SDK. Can we use `use` instead?

File: docs/java/index.md [hunk 4]
```diff
@@ -23,36 +19,31 @@ Our SDKs give you access to our [APIs](/developers/docs/concepts/apis/) and thei
...
+[View on GitHub](https://github.com/contentful/contentful.java)<br/>
 [API reference](https://contentful.github.io/contentful.java/)
 
 ### Content Management API SDK
 
-This SDK interacts with the Content Management API, and allows you to create, edit, manage, and publish content. The API also offers tools for managing editorial teams and cooperation.
-
-[View on GitHub](https://github.com/contentful/contentful-management.java)
-
+This SDK interacts with the Content Management API, and allows you to create, edit, manage, and publish content. The API also offers tools for managing editorial teams and cooperation.<br/>
```
Comment (mariobodemann) [inline docs/java/index.md]: Similar as before: RP is not part of CMA yet.

File: docs/java/index.md [hunk 5]
```diff
@@ -23,36 +19,31 @@ Our SDKs give you access to our [APIs](/developers/docs/concepts/apis/) and thei
...
+[View on GitHub](https://github.com/contentful/contentful-management.java)<br/>
 [API reference](https://contentful.github.io/contentful-management.java/)
 
 ## Tutorials
 
-### API demo
+### Create and deploy a Java application using Contentful
 
-[This guide](/developers/api-demo/java/) is the perfect starting point to learn how to make calls to Contentful APIs, explains what responses look like, and suggest next steps.
+[This guide](https://github.com/contentful/product-catalogue-java) shows you how to create and deploy a Java + Spark app that uses Contentful.
```
Comment (mariobodemann) [inline docs/java/index.md]: `Spark` → `Spring Boot`, `app` → `Website`?

File: docs/java/index.md [hunk 6]
```diff
@@ -23,36 +19,31 @@ Our SDKs give you access to our [APIs](/developers/docs/concepts/apis/) and thei
...
-[This guide](https://github.com/contentful/product-catalogue-java) shows you how to create and deploy a Java and Spring Boot app that uses Contentful.
+[This guide](/developers/api-demo/java/) is the perfect starting point to learn how to make calls to Contentful APIs, explains what responses look like, and suggest next steps.
 
 ## Tools and integrations
 
 ### Vault
 
-A library to simplify persisting data retrieved from Contentful in SQLite.
-
+A library to simplify persisting data retrieved from Contentful in SQLite.<br/>
```
Comment (mariobodemann) [inline docs/java/index.md]: Vault is usable mostly for Android. Java is not a supported target.

**Comment by ChrisChinchilla** [comment]
@realityking Really? I might see if we can get that fixed. I'll check when I get to the preview stage.

File: docs/android/index.md [hunk 7]
```diff
@@ -63,14 +58,12 @@ This SDK interacts with the Content Management API, and allows you to create, ed
 
 ### IntelliJ / Android Studio plugin
 
-This plugin generates model classes automatically from your Contentful content model.
-
+This plugin generates model classes automatically from your Contentful content model.<br/>
```
Comment (ChrisChinchilla) [inline docs/android/index.md]: @mariobodemann Does it still work?

Comment (mariobodemann) [inline docs/android/index.md]: I will check it right now ... ;) I haven't touched it at all ....

Comment (mariobodemann) [inline docs/android/index.md]: It still works, it is a bit rough on the edges, but I can generate my models from inside intellij .... \o>

(The installation might be a bit tricky, because you need to download a zip, and then point the plugin to install this zip, but from there it seems doable)
